kafka/conn.py

```diff
@@ -778,7 +778,8 @@ def filter(self, record):
         log.addFilter(log_filter)
 
         test_cases = [
-            ((0, 10), ApiVersionRequest[0]()),
+            ((0, 10, 1), MetadataRequest[2]([])),
```

Comment:
ah yes. forgot about this little wart of kafka brokers. so it turns out that kafka < 0.10 will silently ignore incorrect metadata api version requests (or at least not close the socket as expected). The socket behavior is different between py2 and py3, causing KafkaConnection._recv() to block indefinitely on py2.

So we can't use MetadataRequest[2] to check version and will need to implement the smarter ApiVersion approach to avoid breaking auto version checks on older brokers.